src/meta-assistant/forecastapi/forecast-api.ts

```diff
@@ -0,0 +1,56 @@
+import { NonProjectTime, TimeRegistrations } from "src/types/meta-assistant/index";
+import fetch, { Headers } from "node-fetch";
+
+/**
+ * Namespace for forecast-api
+ */
+namespace ForecastApiUtilities {
+  const header ={
+    "X-FORECAST-API-KEY": process.env.FORECAST_API_KEY
+  };
+
+  const headers = new Headers(header);
+
+  /**
+   * Gets non project time types from forecast
+   *
+   * @returns All non project times where is_internal_time is false
+   */
+  export const getNonProjectTime = async (): Promise<NonProjectTime[]> => {
+    try {
+      const request: any = await fetch(`${process.env.FORECAST_BASE_URL}/v1/non_project_time`, { headers: headers });
+      const result: any = await request.json();
+
+      if (request.status !== 200) throw new Error(result.message);
+
+      return result.filter(nonProjectTime => !nonProjectTime.is_internal_time);
+    } catch (error) {
+      throw new Error(`Error while loading non project times, ${error.message}`);
+    }
+  };
+
+  /**
+   * Get all allocations after yesterday
+   *
+   * @param dayBeforeYesterday the day before yesterday
+   * @returns all allocations after yesterday
+   */
+  export const getTimeRegistrations = async (dayBeforeYesterday: string): Promise<TimeRegistrations[]> => {
```

Comment:
Why the day is passed as dayBeforeYesterday, not e.g. as date?

Reply:
Why the date is passed as string?

Reply:
Parameter name has been updated. Dates are currently set as strings in the getPreviousTwoWorkdays util as this is the typing they are used in throughout the code. I could update this now or perhaps it could be left as part of a trainee task when this merge is complete (so we can prioritize resolving the weeklyMessage bug that currently exists in production?